query/src/main/scala/filodb/query/exec/BinaryJoinExec.scala

```diff
@@ -72,14 +73,27 @@ final case class BinaryJoinExec(queryContext: QueryContext,
                               querySession: QuerySession): Observable[RangeVector] = {
     val span = Kamon.currentSpan()
     val taskOfResults = childResponses.map {
-      case (QueryResult(_, _, result, _, _, _), _)
-        if (
-          result.size  > queryContext.plannerParams.enforcedLimits.joinQueryCardinality &&
-          cardinality == Cardinality.OneToOne
-        ) => throw new BadQueryException(s"The join in this query has input cardinality of ${result.size} which" +
-          s" is more than limit of ${queryContext.plannerParams.enforcedLimits.joinQueryCardinality}." +
-          s" Try applying more filters or reduce time range.")
-      case (QueryResult(_, _, result, _, _, _), i) => (result, i)
+      tuple : (QueryResult, Int) => {
```

Comment:
Wondering if it is a good idea to separate this logic out as a function and add UTs(for the logger and exception) ?